src/main/java/server/BusAlert/BusAlertService.java

```diff
@@ -1,10 +1,142 @@
 package server.BusAlert;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import server.BusAlert.Location.LocationRequest;
+import server.BusAlert.Rider.Rider;
+import server.BusAlert.Rider.RiderService;
+import server.BusAlert.Route.Route;
+import server.BusAlert.Route.RouteService;
+import server.BusAlert.Stop.Stop;
+import server.BusAlert.Stop.StopService;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 @Service
 public class BusAlertService {
 
+    @Autowired
+    private RouteService routeService;
+
+    @Autowired
+    private RiderService riderService;
+
+    @Autowired
+    private StopService stopService;
+
+    public void receiveGPS(LocationRequest locationRequest){
+
+        if(routeService.getRouteByShortCode(locationRequest.getRouteId()) != null) {
+
+            routeService
+                    // use the routeService to pull the reoute from the ShortCode
+                    // which was passed inside the locationRequest
+                    .getRouteByShortCode(locationRequest.getRouteId())
+
+                    // From the Route object that matches the Route's ShortCode
+                    // get a List of Stops
+                    .getStops()
+
+                    // turn this into parallel streams for better processing
+                    .parallelStream()
+
+                    // call helper method atStop which says if GPS coordinates passed
+                    // in with LocationRequest is with 0.5 km of each stop
+                    // only the True response remain in the stream
+
+                    .filter(stop -> checkOtherStops(locationRequest, stop))
+                    .forEach(stop -> stop
+                            .getRoute()
+                            .getStops()
+                            .get(stop.getRoute().getStops().indexOf(stop) + 2)
+                            .getRiders()
+                            .parallelStream()
+                            .map(rider ->
+                                    riderService
+                                            .notifyRider(rider)
+                            )
+                            .filter(Objects::nonNull)
```

Comment:
this is slick, but `notififyRider` already prints an error message, wouldn't running `failedRiderNotifications` just duplicate our logs?